### DISCUSSION TITLE
'MockValSer' object cannot be converted to 'SchemaSerializer'

### DISCUSSION THREAD

**Discussion by vigneshmanick** [pydantic/pydantic#7710] [Question]
Hello,

I am using pydantic v2.4.2 and when using `model_dump()` i get the following error.

Is there a way to have more stacktrace/add debug option so that the model/location causing the issue  could be identified?

```

Traceback (most recent call last):
  File "test_recursive.py", line 4, in <module>
    dump = model.model_dump()
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
File "pydantic/main.py", line 308, in model_dump
    return self.__pydantic_serializer__.to_python(
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
TypeError: 'MockValSer' object cannot be converted to 'SchemaSerializer'
```

Thanks!

**Comment by rickyyx** [comment]
May I ask why was it closed?

**Reply by vigneshmanick** [reply]
I closed it and created an issue with more details , see https://github.com/pydantic/pydantic/issues/7713

**Comment by anchit2000** [comment]
This issue also happens when the Object is not JSON serializable and you have an HTTP middleware with call_next to log APIs for example. 

For instance, I was returning the ChatCompletion object as is from OpenAI and was getting this error because it is not JSON serializable.
Converting it into a str immediately resolved the issue!  

Parsing outputs to schemas is always a good idea :) 

![image](https://github.com/user-attachments/assets/ca099c39-2d43-4ad9-8051-8e7126fae0cf)
![image](https://github.com/user-attachments/assets/af76285a-38de-44a9-9c91-b15f0437cb2c)
